app/Http/Controllers/StudentEnrollmentController.php

```diff
@@ -0,0 +1,356 @@
+                return response()->json([
+                    'message' => 'Only students can access this endpoint'
+                ], 403);
+            }
+
+            $enrollments = StudentEnrollment::with(['course', 'course.lecturers'])
+                ->where('user_id', $userId)
+                ->where('status', 'active')
+                ->get();
+
+            // Transform data to match frontend expectations
+            $transformedEnrollments = $enrollments->map(function ($enrollment) {
+                return [
+                    'id' => (string) $enrollment->id,
+                    'user_id' => (string) $enrollment->user_id,
+                    'course_id' => (string) $enrollment->course_id,
+                    'semester' => $enrollment->semester,
+                    'tahun_akademik' => $enrollment->tahun_akademik,
+                    'status' => $enrollment->status,
+                    'created_at' => $enrollment->created_at->toISOString(),
+                    'updated_at' => $enrollment->updated_at->toISOString(),
+                    'course' => [
+                        'id' => (string) $enrollment->course->id,
+                        'kode' => $enrollment->course->kode,
+                        'nama' => $enrollment->course->nama,
+                        'semester' => $enrollment->course->semester,
+                        'sks' => $enrollment->course->sks,
+                        'lecturers' => $enrollment->course->lecturers->map(function ($lecturer) {
+                            return [
+                                'id' => (string) $lecturer->id,
+                                'nama' => $lecturer->nama,
+                                'nip' => $lecturer->nip,
+                            ];
+                        })
+                    ]
+                ];
+            });
+
+            return response()->json($transformedEnrollments);
+        } catch (\Exception $e) {
+            Log::error('Error fetching student courses: ' . $e->getMessage());
+            return response()->json([], 200); // Return empty array instead of error
+        }
+    }
+
+    // Enroll student to course
+    public function enrollCourse(Request $request)
+{
+    try {
+        $request->validate([
+            'user_id' => 'required|exists:users,id',
+            'course_id' => 'required|exists:course,id',
+            'semester' => 'required|string'
+        ]);
+
+        // Check if user is mahasiswa
+        $user = User::find($request->user_id);
+        if ($user->role !== 'mahasiswa') {
+            return response()->json([
+                'message' => 'Only students can enroll in courses'
+            ], 403);
+        }
+
+        // Check if already enrolled
+        $existingEnrollment = StudentEnrollment::where('user_id', $request->user_id)
+            ->where('course_id', $request->course_id)
+            ->where('semester', $request->semester)
+            ->where('tahun_akademik', '2024/2025')
```

Comment:
Jangan hardcode `tahun_akademik`
Mungkin seperti ini? (belum tes)
```suggestion
            ->where('tahun_akademik', $request->tahun_akademik ?? '2024/2025')
```